react-native/index.d.ts

```diff
@@ -3161,6 +3161,51 @@ export interface ImageStyle extends FlexStyle, TransformsStyle, ShadowStyleIOS {
     opacity?: number
 }
 
+/*
+    * @see https://github.com/facebook/react-native/blob/master/Libraries/Image/ImageSourcePropType.js
+    */
+interface ImageURISource {
+    /**
+     * `uri` is a string representing the resource identifier for the image, which
+     * could be an http address, a local file path, or the name of a static image
+     * resource (which should be wrapped in the `require('./path/to/image.png')`
+     * function).
+     */
+    uri?: string,
+    /**
+     * `bundle` is the iOS asset bundle which the image is included in. This
+     * will default to [NSBundle mainBundle] if not set.
+     * @platform ios
+     */
+    bundle?: string,
+    /**
+     * `method` is the HTTP Method to use. Defaults to GET if not specified.
+     */
+    method?: string,
+    /**
+     * `headers` is an object representing the HTTP headers to send along with the
+     * request for a remote image.
+     */
+    headers?: {[key: string]: string},
+    /**
+     * `body` is the HTTP body to send with the request. This must be a valid
+     * UTF-8 string, and will be sent exactly as specified, with no
+     * additional encoding (e.g. URL-escaping or base64) applied.
+     */
+    body?: string,
+    /**
+     * `width` and `height` can be specified if known at build time, in which case
+     * these will be used to set the default `<Image/>` component dimensions.
+     */
+    width?: number,
+    height?: number,
+    /**
+     * `scale` is used to indicate the scale factor of the image. Defaults to 1.0 if
+     * unspecified, meaning that one image pixel equates to one display point / DIP.
+     */
+    scale?: number,
+}
```

Comment:
Should the `cache` property be included as well? https://github.com/facebook/react-native/blob/7c9173bc9ef4683d2fe82ad1fcbe81b44c635f99/Libraries/Image/ImageSourcePropType.js#L45-L70